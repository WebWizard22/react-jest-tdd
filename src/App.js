import React, { Component } from 'react';

import logo from './logo.svg';
import './App.css';
import HelloSection from './components/HelloSection';

class App extends Component {
  constructor() {
    super();
    this.state = {
      title: 'REACT IS 4W350M3',
    };
  }

  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>{this.state.title}</h2>
        </div>
        <p className="App-intro">
          This app do something! Let's develop this app using TDD!
        </p>
        <HelloSection username={'John Doe'} />
      </div>
    );
  }
}

export default App;
