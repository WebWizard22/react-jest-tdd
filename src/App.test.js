import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from './App';

describe('<App> component ', () => {
  const appWrapper = shallow(<App />);
  /*
  ============
  OUTPUT TESTS
  ============
  */

  it('renders correctly', () => {
    expect(appWrapper).toHaveLength(1);
  });

  /*
  ============
  STATE TESTS
  ============
  */

  it('have title state', () => {
    const titleState = appWrapper.state('title');
    expect(titleState).toBeDefined();
  });

  it('have title state as a string', () => {
    const titleState = appWrapper.state('title');
    const isAString = typeof titleState === 'string';
    expect(isAString).toBeTruthy();
  });

  /*
  ==============
  SNAPSHOT TESTS
  ==============
  */

  it('renders correctly and matched snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
