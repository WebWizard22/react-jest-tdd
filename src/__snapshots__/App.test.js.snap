// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> component  renders correctly and matched snapshot 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      REACT IS 4W350M3
    </h2>
  </div>
  <p
    className="App-intro"
  >
    This app do something! Let's develop this app using TDD!
  </p>
  <section>
    <h1>
      Hello!
    </h1>
    <h2>
      Username: 
      John Doe
    </h2>
    <p>
      I have some todos below
    </p>
    <ul>
      <li>
        <span>
          My first todo
        </span>
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          DELETE
        </button>
      </li>
    </ul>
  </section>
</div>
`;
